src/registry/mod.rs

```diff
@@ -45,30 +45,25 @@ pub fn write_lock<'a>() -> RwLockWriteGuard<'a, RegMap> {
     REGISTRY.write().unwrap()
 }
 
-/// Gets a Json object from a registry key
-pub fn get_json<K>(name: &K) -> Option<(AccessFlags, Arc<Json>)>
+/// Gets a Lua object from a registry key
+pub fn get_lua<K>(name: &K) -> Option<(AccessFlags, Arc<AnyLuaValue>)>
 where String: Borrow<K>, K: Hash + Eq + Display {
-    trace!("get_json: {}", *name);
+    trace!("get_lua: {}", *name);
     let ref reg = *read_lock();
-    if let Some(val) = reg.get(name) {
-        Some((val.flags(), val.get_json()))
-    }
-    else {
-        None
-    }
+    reg.get(name).map(|val| (val.flags(), val.get_lua()))
 }
 
-/// Gets an object from the registry, decoding its internal json
+/// Gets an object from the registry, decoding its internal Lua
 /// representation.
 pub fn get<K, T>(name: &K) -> Result<(AccessFlags, T), RegistryError>
-where T: Decodable, String: Borrow<K>, K: Hash + Eq + Display {
-    let maybe_json = get_json(name);
-    if let Some(json_pair) = maybe_json {
-        let (access, json_arc) = json_pair;
-        let mut decoder = json::Decoder::new(json_arc.deref().to_json());
-        match T::decode(&mut decoder) {
+where T: FromTable, String: Borrow<K>, K: Hash + Eq + Display {
+    let maybe_lua = get_lua(name);
+    if let Some(lua_pair) = maybe_lua {
```

Comment:
Unnecessary `maybe_lua`, perhaps `if let Some(lua_pair) = get_lua(name)` is better?